internal/controller/controllers/agent_controller.go

```diff
@@ -440,13 +467,28 @@ func (r *AgentReconciler) updateStatus(ctx context.Context, log logrus.FieldLogg
 			agent.Status.ValidationsInfo = newValidationsInfo
 		}
 
-		if h.Progress != nil && h.Progress.CurrentStage != "" {
-			if isNoneDay2Rebooting, err = r.isDay2NonePlatformHostRebooting(ctx, agent, h); err != nil {
-				log.WithError(err).Errorf("Failed to find if agent %s/%s belongs to none platform cluster and is rebooting", agent.Namespace, agent.Name)
-				return ctrl.Result{RequeueAfter: defaultRequeueAfterOnError}, nil
+		// Determine whether or not CSRs should be automatically signed.
+		if cluster != nil {
+			if cluster.UserManagedNetworking == nil {
+				// We look at the database cluster object to check for UserManagedNetworking rather than look at
+				// the equivalent AgentClusterInstall field because that field the latter does not get
+				// properly set in Single Node clusters
+				log.Error("UserManagedNetworking was not set, cannot determine whether or not to auto approve CSRs")
```

Comment:
nil UserManagedNetworking should be assumed as false.  Anyway, it is not an error